packages/ui/Scrollable/Scrollable.js

```diff
@@ -64,710 +64,499 @@ const TouchableDiv = ForwardRef({prop: 'ref'}, Touchable('div'));
 			animate: false,
-			overscrollEffect: this.props.overscrollEffectOn.drag
+			overscrollEffect: props.overscrollEffectOn.drag
 		});
 	}
 
-	onDragEnd = (ev) => {
+	function onDragEnd (ev) {
 		if (ev.type === 'dragend') return;
 
-		this.isDragging = false;
+		variables.current.isDragging = false;
 
-		forward('onDragEnd', ev, this.props);
-		if (this.flickTarget) {
-			const {targetX, targetY, duration} = this.flickTarget;
+		forward('onDragEnd', ev, props);
+		if (variables.current.flickTarget) {
+			const {targetX, targetY, duration} = variables.current.flickTarget;
 
-			this.lastInputType = 'drag';
+			variables.current.lastInputType = 'drag';
 
-			this.isScrollAnimationTargetAccumulated = false;
-			this.start({targetX, targetY, duration, overscrollEffect: this.props.overscrollEffectOn.drag});
+			variables.current.isScrollAnimationTargetAccumulated = false;
+			start({targetX, targetY, duration, overscrollEffect: props.overscrollEffectOn.drag});
 		} else {
-			this.stop();
+			stop();
 		}
 
-		if (this.overscrollEnabled) { // not check this.props.overscrollEffectOn.drag for safety
-			this.clearAllOverscrollEffects();
+		if (variables.current.overscrollEnabled) { // not check props.overscrollEffectOn.drag for safety
+			clearAllOverscrollEffects();
 		}
 
-		this.flickTarget = null;
+		variables.current.flickTarget = null;
 	}
 
-	onFlick = (ev) => {
-		const {direction} = this.props;
+	function onFlick (ev) {
+		const {direction} = props;
 
-		this.flickTarget = this.animator.simulate(
-			this.scrollLeft,
-			this.scrollTop,
-			(direction !== 'vertical') ? this.getRtlX(-ev.velocityX) : 0,
+		variables.current.flickTarget = variables.current.animator.simulate(
+			variables.current.scrollLeft,
+			variables.current.scrollTop,
+			(direction !== 'vertical') ? getRtlX(-ev.velocityX) : 0,
 			(direction !== 'horizontal') ? -ev.velocityY : 0
 		);
 
-		if (this.props.onFlick) {
-			forward('onFlick', ev, this.props);
+		if (props.onFlick) {
+			forward('onFlick', ev, props);
 		}
 	}
 
-	calculateDistanceByWheel (deltaMode, delta, maxPixel) {
+	function calculateDistanceByWheel (deltaMode, delta, maxPixel) {
 		if (deltaMode === 0) {
-			delta = clamp(-maxPixel, maxPixel, ri.scale(delta * this.scrollWheelMultiplierForDeltaPixel));
+			delta = clamp(-maxPixel, maxPixel, ri.scale(delta * variables.current.scrollWheelMultiplierForDeltaPixel));
 		} else if (deltaMode === 1) { // line; firefox
-			delta = clamp(-maxPixel, maxPixel, ri.scale(delta * this.pixelPerLine * this.scrollWheelMultiplierForDeltaPixel));
+			delta = clamp(-maxPixel, maxPixel, ri.scale(delta * variables.current.pixelPerLine * variables.current.scrollWheelMultiplierForDeltaPixel));
 		} else if (deltaMode === 2) { // page
 			delta = delta < 0 ? -maxPixel : maxPixel;
 		}
 
 		return delta;
 	}
 
-	onWheel = (ev) => {
-		if (this.isDragging) {
+	const onWheel = useCallback((ev) => {
```

Comment:
I found `props` inside this callback and there is no dependency for `props`.

Reply:
I added `props` for dependency. Is this what you mean?